rust/src/mime/detect.rs

```diff
@@ -97,3 +98,20 @@ pub unsafe extern "C" fn SCDetectMimeEmailGetDataArray(
     }
     return 0;
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn SCDetectMimeEmailGetBodyMd5(
+    ctx: &MimeStateSMTP, buffer: *mut *const u8, buffer_len: *mut u32,
+) -> u8 {
+    if ctx.md5_state == MimeSmtpMd5State::MimeSmtpMd5Completed {
```

Comment:
Do not we have a SMTP state to do this ?

Reply:
Hmm, I didn't understand the question @catenacyber

Reply:
App-layer transactions have different states/progress and keywords/buffers register a min progress they can act one

If you register the buffer at the right state/progress, you should not need this